src/vs/workbench/contrib/terminal/common/remoteTerminalChannel.ts

```diff
@@ -195,13 +199,19 @@ export class RemoteTerminalChannelClient {
 		return await this._channel.call<ICreateTerminalProcessResult>('$createProcess', args);
 	}
 
+	requestAdoptInstance(workspaceId: string, instanceId: number): Promise<void> {
+		return this._channel.call('$requestAdoptInstance', [workspaceId, instanceId]);
+	}
+	acceptInstanceForAttachment(persistentProcessId: number): Promise<void> {
+		return this._channel.call('$acceptInstanceForAttachment', [persistentProcessId]);
+	}
 	attachToProcess(id: number): Promise<void> {
 		return this._channel.call('$attachToProcess', [id]);
 	}
 	detachFromProcess(id: number): Promise<void> {
 		return this._channel.call('$detachFromProcess', [id]);
 	}
-	listProcesses(): Promise<IProcessDetails[]> {
+	listProcesses(all?: boolean): Promise<IProcessDetails[]> {
```

Comment:
currently seeing if I can remove this argument all over and instead just check `this._processIdToAttach`

Reply:
I think that's dangerous because then we won't know if it's undefined bc of a race condition or bc it's not a transfer between windows

Reply:
can change all everywhere to something more descriptive though